### ISSUE TITLE
Use torch ShuffleDataset 

### ISSUE THREAD

**Issue by eddyxu** [eto-ai/rikai#81]
https://github.com/pytorch/pytorch/pull/45290

Offers shuffling for iterative datasets.

**Comment by eddyxu** [comment]
https://pytorch.org/docs/master/data.html#torch.utils.data.BufferedShuffleDataset
